Feat/user

DOGYUN0903 opened the pull request:
- 공통 응답객체(ApiResponse)
- 공통 예외처리
- 로그인
- 회원가입

DOGYUN0903 committed:
feat: 공통 응답 객체 API 구현

DOGYUN0903 committed:
feat: 공통으로 사용되는 BaseEntity 구현

DOGYUN0903 committed:
feat: User 엔티티 구현 및 권한 관련 enum, 에러코드 추가

DOGYUN0903 committed:
feat: 공통 비즈니스 예외 처리 로직 및 핸들러 구현

DOGYUN0903 committed:
feat: User의 유효하지 않은 권한 예외 클래스 추가

DOGYUN0903 committed:
feat: Auditing 적용

DOGYUN0903 committed:
feat: JWT 토큰 의존성 추가 및 구현

DOGYUN0903 committed:
feat: Spring Security 구현

DOGYUN0903 committed:
feat: 회원가입 및 로그인 API 구현

DOGYUN0903 committed:
feat: 커스텀 예외 구현 및 전역 예외 핸들러에 @Valid, 공백값에 대한 오류 추가

DOGYUN0903 committed:
feat: 로그인 시 회원 탈퇴 여부 확인 및 커스텀 예외 구현

DOGYUN0903 committed:
refactor: User 엔티티 필드에 nullable = false 제약 추가

DOGYUN0903 committed:
delete: 사용하지 않는 커스텀 예외 클래스 삭제

balsohn reviewed (commented):
1. BasEntity 오타수정
- createAt -> createdAt 
- public -> private

2. UserEntity 테이블명
- user -> users

3. AuthController 응답형식
- ApiResponse 사용

4. JwtUtil.createToken()
- userRole 파라미터 추가?

balsohn committed:
refactor: JWT 필터 책임 분리

kimyongjun0129 commented:
지호님이 말씀해주신 거 외에는 괜찮은거 같습니다.

DOGYUN0903 committed:
refactor: 피드백 받은 것들 수정

- BaseEntity 오타 수정 createAt -> createdAt, updatedAt을 private으로 수정
- UserEntity에 users 테이블 명 추가
- AuthController에 공통응답객체 사용
- JwtUtil.createToken에 userRole 파라미터 추가
- UserException, AuthException 파일 생성 후 커스텀 예외 관리

DOGYUN0903 committed:
Merge remote-tracking branch 'origin/feat/user' into feat/user
